tools/quadrados/src/entry.cpp

```diff
@@ -4,32 +4,89 @@
 #include "tools.hpp"
 
 /// Prints the help message of the program.
-int runHelp(int /*argc*/, char** /*argv*/)
+int runHelp(int /*argc*/, char** /*argv*/, GlobalArgs& /*ga*/)
 {
-    std::cerr << "Usage: quadrados <TOOL>" << std::endl;
+    std::cerr << "Usage: quadrados [OPTIONS] <TOOL>" << std::endl;
     std::cerr << "Tools:" << std::endl;
     for (const auto& tool : Tools)
     {
         std::cerr << "  " << tool.name << std::endl;
     }
+    std::cerr << "Options:" << std::endl;
+    std::cerr << "  -D <install_directory> path to install directory" << std::endl;
+
     return 0;
 }
 
+/*
+/   Checks if a string is a tool
+*/
+bool isTool(char* param)
+{
+    for (const auto& tool : Tools)
+    {
+        if (tool.name == param)
+        {
+
+            return true;
+        }
+    }
+
+    return false;
+}
+
+/*
+/   Returns the number of parsed args
+/   and updates the GlobalArgs struct if needed
+*/
+int parseGlobalArgs(int argc, char** argv, GlobalArgs& ga)
+{
+    int nParsed = 0;
+    for (int i = 1; i < argc; i++)
+    {
+        std::string arg = argv[i];
+        if (arg == "-D" && i + 1 < argc)
+        {
+            ga.installDir = argv[++i];
+            nParsed += 2;
+        }
+        else if (arg == "-D")
+        {
+            return -1;
+        }
+        else if (isTool(argv[i]))
+        {
+            return nParsed;
+        }
```

Comment:
```suggestion
        }
        else
        {
          std::cerr << "Unknown option " << argv[i] << std::endl;
          return -1;
        }
```
Perhaps might make debugging future errors easier